KakapoExample/KakapoTests/JSONAPITests.swift

```diff
@@ -0,0 +1,270 @@
+        }
+        
+        describe("JSON API serializer") {
+            it("should serialize data") {
+                let object = json(JSONAPISerializer(user))
+                let data = object["data"].dictionaryValue
+                expect(data.count).toNot(equal(0))
+            }
+            
+            it("should serialize data from Array") {
+                let object = json(JSONAPISerializer([user]))
+                let data = object["data"].arrayValue
+                expect(data.count).toNot(equal(0))
+            }
+        }
+        
+        describe("JSON API Entity Serialization") {
+            
+            it("should serialize the attributes") {
+                let object = json(user)
+                let attributes = object["attributes"]
+                expect(attributes["id"].string).to(beNil())
+                expect(attributes["name"].stringValue).to(equal("Alex"))
+                expect(attributes.dictionaryValue.count).to(equal(1))
+            }
+            
+            it("should serialize the id") {
+                let object = json(user)
+                expect(object["id"].stringValue).to(equal("11"))
+            }
+            
+            it("should serialize the type") {
+                let object = json(user)
+                expect(object["type"].stringValue).to(equal("user"))
+            }
+            
+            it("should serialize an arrays of non-JSONAPIEntities as an attribute") {
+                let object = json(Post(id: "11", relatedPostIds: [1, 2, 3]))
+                let relatedPostIds = object["attributes"]["relatedPostIds"].arrayValue
+                expect(relatedPostIds.count).to(equal(3))
+                expect(relatedPostIds[0]).to(equal(1))
+                expect(relatedPostIds[1]).to(equal(2))
+                expect(relatedPostIds[2]).to(equal(3))
+            }
+            
+            it("should serialize an array of JSONAPIEntities") {
+                let objects = json([user, user]).arrayValue
+                for object in objects {
+                    expect(object["attributes"]["name"].stringValue).to(equal("Alex"))
+                }
+                
+                expect(objects.count).to(equal(2))
+            }
+            
+            it("should only serialize actual attributes into attributes") {
+                let lonelyMax = User(id: "11", name: "Max", dog: dog, cats: [])
+                let object = json(lonelyMax)
+                let attributes = object["attributes"].dictionaryValue
+                expect(attributes.count).to(equal(1)) // only name should be here, no id, no cats
+                expect(attributes["name"]).to(equal("Max"))
+                expect(attributes["cats"]).to(beNil())
+                expect(attributes["id"]).to(beNil())
+            }
+            
+            it("should fail to serialize CustomSerializable entities") {
+                // TODO: discuss because this might be unexpected
```

Comment:
Ok then now this is the expected behavior isn't it? If you override `customSerialize` you will get back your own implementation.